src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -19,224 +19,351 @@
 package ch.njol.skript.registrations;
 
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import com.google.common.collect.ImmutableList;
+
 import ch.njol.skript.Skript;
 import ch.njol.skript.classes.Converter;
 import ch.njol.skript.expressions.base.EventValueExpression;
 import ch.njol.skript.util.Getter;
 
 /**
- * @author Peter Güttinger
+ * The main class for registering and getting anything relating to event values.
  */
 public class EventValues {
-	
-	private EventValues() {}
-	
-	private final static class EventValueInfo<E extends Event, T> {
-		
-		public final Class<E> event;
-		public final Class<T> c;
-		public final Getter<T, E> getter;
-		@Nullable
-		public final Class<? extends E>[] excludes;
+
+	private final static class EventValueInfo<E extends Event, T> implements Comparable<EventValueInfo<?, ?>> {
+
+		private final Class<? extends E>[] excludes;
+
 		@Nullable
-		public final String excludeErrorMessage;
-		
-		public EventValueInfo(Class<E> event, Class<T> c, Getter<T, E> getter, @Nullable String excludeErrorMessage, @Nullable Class<? extends E>[] excludes) {
-			assert event != null;
-			assert c != null;
+		private final String excludeErrorMessage;
+
+		private final Getter<T, E> getter;
+		private final Class<E> event;
+		private final Class<T> type;
+
+		private int priority;
```

Comment:
*just an idea/thought*

With the Structure API, we added a custom Priority class that implements comparable. It essentially provides greater control over the comparison process if desired (may be useful for addons too)

See https://github.com/SkriptLang/Skript/blob/ed3487a7586aaafe5f944d56f57cbd77850ef50f/src/main/java/ch/njol/skript/lang/structure/Structure.java#L71

@TPGamesNL 

Do you think something like this could be useful here? Or are int priority likely enough.

Reply:
Done and moved outside of Structure class.